### DISCUSSION TITLE
Create an API endpoint in the Nextjs Subscription Project

### DISCUSSION THREAD

**Discussion by angadveeersb** [vercel/community#3226] [Help] [answered]
### Summary

So I've been using the nextjs-subscription-payments project from vercel and I wanted to allow my customers to make API requests to the /api/get directory. It would require them to pass an apiKey param which is just their customer ID for simplification. example `https://example.com/api/get?apiKey=xxxxx` I don't know how to modify the project to do this. I have also created sql functions in the supabase dashboard to retrieve the price_id of the customers to see which tier subscription they have to limit access to the api. 

```
create or replace function get_price_id_from_userid (user_id text) returns text language sql as $$
  select price_id from subscriptions where user_id = subscriptions.user_id;
$$;
```
when I run get_price_id_from_userid('MYAPIKEY')  in the dashboard it returns the price_id. But how do I integrate this into my project? really appreciate, thanks!

### Example

https://github.com/vercel/nextjs-subscription-payments

### Steps to Reproduce

Please look at the repo

**Comment by amyegan** [comment, marked as answer]
Hi @angadveeersb! You can learn more about adding api routes in the Next.js Route Handlers documentation at https://nextjs.org/docs/app/building-your-application/routing/router-handlers

You can get the query parameters from the request URL as `searchParams`, like in this [Dynamic Route Handlers](https://nextjs.org/docs/app/building-your-application/routing/router-handlers#dynamic-route-handlers) example.
